grafana-plugin/src/pages/settings/SettingsPage.tsx

```diff
@@ -21,18 +23,23 @@ import styles from './SettingsPage.module.css';
 
 const cx = cn.bind(styles);
 
-interface SettingsPageProps {
-  store: RootBaseStore;
-}
+interface SettingsPageProps extends AppRootProps, WithStoreProps {}
 interface SettingsPageState {
   activeTab: string;
 }
 
 @observer
 class Settings extends React.Component<SettingsPageProps, SettingsPageState> {
-  state: SettingsPageState = {
-    activeTab: SettingsPageTab.MainSettings.key, // should read from route instead
-  };
+  constructor(props: SettingsPageProps) {
+    super(props);
+    const {
+      query: { tab }, // eslint-disable-line
```

Comment:
I suggest to not ignore deprecation warning from @grafana/data. We have `LocationHelper.getQueryParam`, can we use it instead?

Minor: This component is super small, maybe worth to switch it to functional component